ingestion_server/test/integration_test.py

```diff
@@ -66,575 +65,650 @@ def handle_task_callback():
-        self.assertEqual(count, 5000, msg)
-
-    @pytest.mark.order(after="test_promote_images")
-    def test_filtered_image_index_creation(self):
-        self._create_and_populate_filtered_index("image", "integration", "integration")
-
-    @pytest.mark.order(after="test_promote_audio")
-    def test_filtered_audio_index_creation(self):
-        self._create_and_populate_filtered_index("audio", "integration", "integration")
-
-    @pytest.mark.order(after="test_filtered_image_index_creation")
-    def test_point_filtered_image_alias(self):
-        self._point_alias("image", "integration-filtered", "image-filtered")
-
-    @pytest.mark.order(after="test_filtered_audio_index_creation")
-    def test_point_filtered_audio_alias(self):
-        self._point_alias("audio", "integration-filtered", "audio-filtered")
-
-    @pytest.mark.order(
-        after=["test_point_filtered_audio_alias", "test_point_filtered_image_alias"]
+@pytest.mark.order(after="test_image_ingestion_succeeds")
+def test_task_count_after_one():
+    res = requests.get(f"{ingestion_server}/task")
+    res_json = res.json()
+    msg = "There should be one task in the task list now."
+    assert 1 == len(res_json), msg
+
+
+@pytest.mark.order(after="test_task_count_after_one")
+def test_audio_ingestion_succeeds(setup_fixture):
+    _ingest_upstream(
+        setup_fixture["cb_queue"],
+        setup_fixture["downstream_db"],
+        "audio",
+        "integration",
+    )
+
+
+@pytest.mark.order(after="test_audio_ingestion_succeeds")
+def test_task_count_after_two():
+    res = requests.get(f"{ingestion_server}/task")
+    res_json = res.json()
+    msg = "There should be two tasks in the task list now."
+    assert 2 == len(res_json), msg
+
+
+@pytest.mark.order(after="test_task_count_after_two")
+def test_promote_images(setup_fixture):
+    _promote(setup_fixture["cb_queue"], "image", "integration", "image-main")
+
+
+@pytest.mark.order(after="test_promote_images")
+def test_promote_audio(setup_fixture):
+    _promote(setup_fixture["cb_queue"], "audio", "integration", "audio-main")
+
+
+@pytest.mark.order(after="test_promote_audio")
+def test_upstream_indexed_images():
+    """
+    Check that the image data has been successfully indexed in Elasticsearch.
+
+    The number of hits for a blank search should match the size of the loaded mock
+    data.
+    """
+
+    es = _get_es()
+    es.indices.refresh(index="image-integration")
+    count = es.count(index="image-integration")["count"]
+    msg = "There should be 5000 images in Elasticsearch after ingestion."
+    assert count == 5000, msg
+
+
+@pytest.mark.order(after="test_upstream_indexed_images")
+def test_upstream_indexed_audio():
+    """
+    Check that the audio data has been successfully indexed in Elasticsearch.
+
+    The number of hits for a blank search should match the size of the loaded mock
+    data.
+    """
+
+    es = _get_es()
```

Comment:
Maybe `_get_es` could also be converted to a fixture? It's used as such, but I don't know what the specific benefit would be either way, other than that it _looks_ like a fixture to me.

Reply:
Yes, I have converted `_get_es` to `sample_es` as a fixture.